api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
-    members_to_add = None
-    if 'person_ids' in request.json.keys():
-        members_to_add = request.json['person_ids']
-        del request.json['person_ids']
+def update_group_type(group_type_id):
+    group_type_schema = GroupTypeSchema()
 
     try:
-        valid_group = group_schema.load(request.json)
+        valid_attributes = group_type_schema.load(request.json, partial=True)
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    new_group = Group(**valid_group)
+    return modify_entity(GroupType, group_type_schema, group_type_id, valid_attributes)
 
-    if db.session.query(Manager).filter_by(id=new_group.manager_id).first() is None:
-        return jsonify(msg="Manager not found"), 404
 
-    db.session.add(new_group)
+@groups.route('/group-types/<int:group_type_id>', methods=['DELETE'])
+@jwt_required
+def delete_group_type(group_type_id):
+    group_type = db.session.query(GroupType).filter_by(id=group_type_id).first()
 
-    today = datetime.datetime.today().strftime('%Y-%m-%d')
+    if not group_type:
+        return jsonify(f"GroupType with id #{group_type_id} does not exist."), 404
 
-    if members_to_add is not None:
-        for member in members_to_add:
-            new_member = generate_member(new_group.id, member, today, True)
-            db.session.add(new_member)
+    db.session.delete(group_type)
+    db.session.commit()
 
-    # Add group_overseer role to the existing manager account -> if they have an account
-    group_overseer = db.session.query(Role).filter_by(name_i18n="role.group-overseer").first()
-    subq = db.session.query(Manager.person_id).filter_by(id=new_group.manager_id).subquery()
-    manager_account = db.session.query(Person).filter(Person.id.in_(subq)).first()
+    # 204 codes don't respond with any content
+    return "Deleted successfully", 204
 
-    if manager_account:
-        manager_roles = manager_account.roles
-        if group_overseer and group_overseer not in manager_roles:
-            manager_roles.append(group_overseer)
-            setattr(manager_account, 'roles', manager_roles)
-            print("adding group overseer role", end='\n\n\n')
+# ---- Group
 
-    # db.session.add(manager_account)
-    db.session.commit()
-    return group_dump(new_group), 201
+group_schema = GroupSchema()
+
+@groups.route('/groups', methods=['POST'])
+@jwt_required
+def create_group():
+    try:
+        valid_group = group_schema.load(request.json, partial=['active'])
+    except ValidationError as err:
+        return jsonify(err.messages), 422
+
+    if 'active' not in valid_group:
+        valid_group['active'] = True
+
+    new_group = Group(**valid_group)
+
+    db.session.add(new_group)
+
+    try:
+        db.session.commit()
+    # when the foreign key given is invalid
+    except IntegrityError:
+        return jsonify('Payload contains an invalid foreign key'), 404
+
+    return jsonify(group_schema.dump(new_group)), 201
 
 
 @groups.route('/groups')
```

Comment:
Add `methods=['GET']`, here and throughout where appropriate.

Reply:
will be fixed